runtime/src/cost_tracker.rs

```diff
@@ -24,41 +27,56 @@ pub enum CostTrackerError {
 pub struct CostTracker {
     account_cost_limit: u64,
     block_cost_limit: u64,
+    vote_cost_limit: u64,
     cost_by_writable_accounts: HashMap<Pubkey, u64>,
     block_cost: u64,
+    vote_cost: u64,
     transaction_count: u64,
 }
 
 impl Default for CostTracker {
     fn default() -> Self {
-        CostTracker::new(MAX_WRITABLE_ACCOUNT_UNITS, MAX_BLOCK_UNITS)
+        CostTracker::new(MAX_WRITABLE_ACCOUNT_UNITS, MAX_BLOCK_UNITS, MAX_VOTE_UNITS)
     }
 }
 
 impl CostTracker {
-    pub fn new(account_cost_limit: u64, block_cost_limit: u64) -> Self {
+    pub fn new(account_cost_limit: u64, block_cost_limit: u64, vote_cost_limit: u64) -> Self {
         assert!(account_cost_limit <= block_cost_limit);
+        assert!(vote_cost_limit <= block_cost_limit);
         Self {
             account_cost_limit,
             block_cost_limit,
+            vote_cost_limit,
             cost_by_writable_accounts: HashMap::with_capacity(WRITABLE_ACCOUNTS_PER_BLOCK),
             block_cost: 0,
+            vote_cost: 0,
             transaction_count: 0,
         }
     }
 
     // bench tests needs to reset limits
-    pub fn set_limits(&mut self, account_cost_limit: u64, block_cost_limit: u64) {
+    pub fn set_limits(
+        &mut self,
+        account_cost_limit: u64,
+        block_cost_limit: u64,
+        vote_cost_limit: u64,
+    ) {
         self.account_cost_limit = account_cost_limit;
         self.block_cost_limit = block_cost_limit;
+        self.vote_cost_limit = vote_cost_limit;
     }
 
     pub fn would_transaction_fit(
         &self,
-        _transaction: &Transaction,
+        transaction: &Transaction,
         tx_cost: &TransactionCost,
     ) -> Result<(), CostTrackerError> {
-        self.would_fit(&tx_cost.writable_accounts, &tx_cost.sum())
+        self.would_fit(
+            &tx_cost.writable_accounts,
+            &tx_cost.sum(),
+            is_simple_vote_transaction(transaction),
```

Comment:
Isn't it better to use the flag we already computed instead of parse the transaction again?

Reply:
Yeah, it is lazier to just call the function :) I'll pipe the packet.meta.is_simple_vote into it.

Reply:
changed to utilize HashedTransaction.is_simple_vote_tx to avoid parsing transaction again